spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'posts GET /index' do
+    before(:example) { get('/users/:user_id/posts') }
+    it 'response is ok for index' do
+      assert_response :ok
+    end
+    it 'render the right page' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'user/posts containes place holder text' do
+      expect(response.body).to match('this is a post index')
+    end
+  end
+  describe 'posts show action tests' do
+    before(:example) { get('/users/:user_id/posts/:id') }
```

Comment:
Good job on the test cases, however, but your Spec file should include tests that ensure the response body includes the correct placeholder text.
Example : `expect(response.body).to include("Welcome to user index page")`

>